### PR TITLE
refactor: examples/peer-and-content-routing

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR refactors the `peer-and-content-routing` example to the refactored async `js-libp2p`.

Needs:

- [x] #490 - this work is on top of the token dialer branch

### CODE REVIEW

File: examples/peer-and-content-routing/1.js [hunk 1]
```diff
@@ -1,77 +1,55 @@
...
+  ])
 
-  parallel([
-    (cb) => node1.dial(node2.peerInfo, cb),
-    (cb) => node2.dial(node3.peerInfo, cb),
-    // Set up of the cons might take time
-    (cb) => setTimeout(cb, 300)
-  ], (err) => {
-    if (err) { throw err }
+  // Set up of the cons might take time
```
Comment (jacobheun) [inline examples/peer-and-content-routing/1.js]:
```suggestion
  // The DHT routing tables need a moment to populate
```

File: examples/peer-and-content-routing/README.md [hunk 2]
```diff
@@ -10,31 +10,28 @@ Content Routing is the category of modules that offer a way to find where conten
...
+    streamMuxer: [ Mplex ],
+    connEncryption: [ SECIO ],
+    // we add the DHT module that will enable Peer and Content Routing
+    dht: KadDHT
+  },
+  config: {
+    dht: {
+      // dht must be enabled
+      enabled: true,
+      kBucketSize: 20
```
Comment (jacobheun) [inline examples/peer-and-content-routing/README.md]: We don't use `kBucketSize` in the code, we should probably just leave it out here. Most users won't need to change the default.

File: examples/peer-and-content-routing/2.js [hunk 3]
```diff
@@ -1,85 +1,59 @@
...
-    const cid = new CID('QmTp9VkYvnHyrqKQuFPiuZkiX9gPcqj6x5LJ1rmWuSySnL')
+  const cid = new CID('QmTp9VkYvnHyrqKQuFPiuZkiX9gPcqj6x5LJ1rmWuSySnL')
+  await node1.contentRouting.provide(cid)
 
-    node1.contentRouting.provide(cid, (err) => {
-      if (err) { throw err }
+  // wait for propagation
+  await delay(300)
 
-      console.log('Node %s is providing %s', node1.peerInfo.id.toB58String(), cid.toBaseEncodedString())
```
Comment (jacobheun) [inline examples/peer-and-content-routing/2.js]: Was this removed intentionally? If so, it should also be removed from the readme. Can you make sure the code in the readme matches what's in the js files

Comment (vasco-santos) [reply examples/peer-and-content-routing/2.js]: I added it back, I removed unintentionally

**Pull request merged**